Incorrect type for post step data items

```console
10:36:09 [ERROR] Exception: Failed creating Slang models for file: '/home/ubuntu/cloud-slang-content/content/io/cloudslang/cloud/openstack/servers/create_server_flow.sl'.
Error transforming source: create_server_flow to a Slang model. For step: update_body_json_with_boot_json - Incorrect type for post step data items.
```

```yaml
- publish:
    - output:
```

execution should fail earlier than the exception we got

--- a/cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java
+++ b/cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java
@@ -192,7 +192,11 @@ public class ExecutableBuilder {
                 errors.addAll(actionModellingResult.getErrors());
                 Action action = actionModellingResult.getAction();
                 executableDependencies = new HashSet<>();
-                systemPropertyDependencies = dependenciesHelper.getSystemPropertiesForOperation(inputs, outputs, results);
+                try {
+                    systemPropertyDependencies = dependenciesHelper.getSystemPropertiesForOperation(inputs, outputs, results);
+                } catch (RuntimeException ex) {
+                    errors.add(ex);
+                }
                 Operation operation = new Operation(
                         preExecutableActionData,
                         postExecutableActionData,

--- a/cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java
+++ b/cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java
@@ -153,7 +153,7 @@ public class ExecutableBuilder {
         String namespace = parsedSlang.getNamespace();
         Map<String, String> imports = parsedSlang.getImports();
         Set<String> executableDependencies;
-        Set<String> systemPropertyDependencies;
+        Set<String> systemPropertyDependencies = null;
         switch (parsedSlang.getType()) {
             case FLOW:
                 List<Map<String, Map<String, Object>>> workFlowRawData = getWorkflowRawData(executableRawData, errors, parsedSlang, execName);
@@ -167,7 +167,11 @@ public class ExecutableBuilder {
                 errors.addAll(validateFlowResultsHaveNoExpression(results, execName));
 
                 executableDependencies = fetchDirectStepsDependencies(workflow);
-                systemPropertyDependencies = dependenciesHelper.getSystemPropertiesForFlow(inputs, outputs, results, workflow.getSteps());
+                try {
+                    systemPropertyDependencies = dependenciesHelper.getSystemPropertiesForFlow(inputs, outputs, results, workflow.getSteps());
+                } catch (RuntimeException ex) {
+                    errors.add(ex);
+                }
                 Flow flow = new Flow(
                         preExecutableActionData,
                         postExecutableActionData,

--- a/cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/PreCompilerErrorsTest.java
+++ b/cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/PreCompilerErrorsTest.java
@@ -556,4 +556,16 @@ public class PreCompilerErrorsTest {
         exception.expectMessage("Step: print_message2 is unreachable");
         throw result.getErrors().get(0);
     }
+
+    @Test
+    public void testNullPublishValue() throws Exception {
+        URI resource = getClass().getResource("/corrupted/null_publish_value.sl").toURI();
+
+        ExecutableModellingResult result = compiler.preCompileSource(SlangSource.fromFile(resource));
+        assertTrue(result.getErrors().size() > 0);
+        exception.expect(RuntimeException.class);
+        exception.expectMessage("For step 'CheckWeather' syntax is illegal.");
+        exception.expectMessage("Could not transform Output : {var_with_null_value=null} since it has a null value.");
+        throw result.getErrors().get(0);
+    }
 }

--- a/cloudslang-compiler/src/test/resources/corrupted/null_publish_value.sl
+++ b/cloudslang-compiler/src/test/resources/corrupted/null_publish_value.sl
@@ -1,0 +1,33 @@
+#   (c) Copyright 2014 Hewlett-Packard Development Company, L.P.
+#   All rights reserved. This program and the accompanying materials
+#   are made available under the terms of the Apache License v2.0 which accompany this distribution.
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+namespace: user.ops
+
+imports:
+  ops: user.ops
+
+flow:
+  name: null_publish_value
+  inputs:
+    - input1
+    - host
+    - port: '22'
+  workflow:
+    - CheckWeather:
+        do:
+          ops.test_op:
+            - city: 'input_1'
+            - port
+            - alla: 'walla'
+        publish:
+          - var_with_null_value:
+        navigate:
+          - SUCCESS: SUCCESS
+          - FAILURE: FAILURE
+  results:
+    - SUCCESS
+    - FAILURE

